edx-solutions/edx-platform
# [YONK-1241] Optimize computing social engagement

This PR changes default queue for `task_compute_social_scores_in_course` to low-priority queue.

## Testing instructions:
### Before testing:
1. Add the following lines to `lms/envs/private.py`:
	```python
	CELERY_ALWAYS_EAGER = False
	BROKER_URL = 'amqp://celery:celery@127.0.0.1:5672'
	LOW_PRIORITY_QUEUE = 'edx.lms.core.low'
	```
1. Set `ENABLE_SOCIAL_ENGAGEMENT` in `lms/env/common.py` to `True`.
1. Remove engagement scores (`StudentSocialEngagementScore.objects.all().delete()`).

### Testing:
1. Start Celery worker (`DJANGO_SETTINGS_MODULE=lms.envs.devstack SERVICE_VARIANT=lms celery -A cms worker -E -l debug`).
1. (Optionally) install and start flower (`pip install flower && DJANGO_SETTINGS_MODULE=lms.envs.devstack SERVICE_VARIANT=lms celery -A lms flower --port=8088`). You can see the progress on localhost:8088.
1. Run the management command `./manage.py lms compute_social_engagement_score -a true --settings=devstack`.
1. Check that social engagement scores are present (`StudentSocialEngagementScore.objects.count()`).
